### PR TITLE
feat: show loading state for user prompts

### PR DESCRIPTION
**Pull request by daniel-mader**
# Description of change
In order to give the user any feedback that something is going on, we want to show some loading spinner.
The `Button` was extended to host a loading spinner, also some other components were updated to "grey out" (disable) when a `loading` state is active. In this PR, we introduce the loading animations when a user prompt is accepted, but other parts of the app are left untouched.

## Links to any relevant issues
- #107 

## How the change has been tested
- iOS (physical)

## Definition of Done checklist
Add an `x` to the boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by SoulaAndrikopoulos — DISMISSED** [review]

File: unime/src/lib/components/atoms/Button.svelte [hunk 1]
```diff
@@ -17,17 +20,33 @@
...
 
 <!-- TODO: does it make sense to pass in the trigger here? -->
-{#if trigger}
-  <button
-    use:melt={trigger}
-    class="h-[48px] w-full rounded-xl px-4 py-2 text-[13px]/[24px] font-medium disabled:opacity-50 {variant_classes}"
-    {disabled}
-    on:click={() => dispatch('click')}>{label}</button
-  >
+{#if !loading}
```
Comment (maiertech) [inline unime/src/lib/components/atoms/Button.svelte:23]: I would put the if to check for the trigger outside and loading-if inside. And then conditionally render the spinner and disable the two types of buttons.

File: unime/src/routes/(app)/scan/+page@.svelte [hunk 2]
```diff
@@ -141,7 +145,14 @@
           </div>
         {/if}
 
-        {#if $state?.dev_mode !== 'Off'}
+        {#if loading}
```
Comment (maiertech) [inline unime/src/routes/(app)/scan/+page@.svelte]: It was a little unexpected that the "Scan a QR Code" heading disappears and then with a delay the spinner on an almost empty screen. I would swap out the camera portion only and overlay the spinner there.

**Review by maiertech — COMMENTED** [review]
I checked all modified routes with NGDIL staging. Huge UX improvement!

File: unime/src/lib/components/atoms/Button.svelte (hunk 1 above)
Comment (daniel-mader) [reply unime/src/lib/components/atoms/Button.svelte:23]: Good point, but let's refactor the `Button` in a future refactoring session 👍

Comment (maiertech) [reply unime/src/lib/components/atoms/Button.svelte:23]: Sounds good!

File: unime/src/routes/(app)/scan/+page@.svelte [hunk 3]
```diff
@@ -126,72 +128,80 @@
...
-      >
-        {#if permissions_nullable && permissions_nullable !== 'granted'}
-          <div class="flex w-3/4 flex-col space-y-4">
-            <div class="flex flex-col items-center rounded-lg bg-rose-100 px-8 py-4 text-rose-500">
-              <CameraSlash class="m-2 h-8 w-8" />
-              <p class="text-center text-[13px]/[24px] font-semibold">{$LL.SCAN.PERMISSION_DENIED()}</p>
+      {#if !scanning && !loading}
+        <!-- This part is only visible when no scanning or loading is happening.
+          Only visible when user has not granted permissions to the camera. -->
+        <div class="relative flex h-full flex-col items-center justify-center space-y-4 bg-silver p-8 dark:bg-navy">
```
Comment (maiertech) [inline unime/src/routes/(app)/scan/+page@.svelte:134]: Might be nitpicking here, but there is usually no reason to use Tailwind's space utilities in combination with a Flex layout. Tailwind has `gap`, `gap-x` and `gap-y`, which corresponds to standard CSS.

Comment (maiertech) [reply unime/src/routes/(app)/scan/+page@.svelte:134]: Do we need `relative` here?

**Review by SoulaAndrikopoulos — COMMENTED** [review]
The loading animation looks smooth, but in the dev environment I can't see what happens after loading (not sure if this is just my device). I also can't cancel the action while it's still loading or go back (just with the 'back' button of the dev tool).

File: unime/src/routes/(app)/scan/+page@.svelte (hunk 3 above)
Comment (maiertech) [reply unime/src/routes/(app)/scan/+page@.svelte:134]: Not crucial to fix given other priorities.

**Review by maiertech — APPROVED** [review]

**Review by nanderstabel — APPROVED** [review]
Very nice improvement!

**Pull request merged**